### DISCUSSION TITLE
re-run jobs

### DISCUSSION THREAD

**Discussion by ericLemanissier** [community/community#26001] [Actions]
<p>Is there a way to re-run individual jobs parts of workflows ? Alternatively is there a way to restart all failed jobs ? Use interface seem to propose only “Re-run all checks” currently</p>

**Comment by chrispat** [comment, marked as answer]
<p>At the moment you are not able to re-run individual jobs or simply restart all failed jobs.  That feature is absolutely on our backlog to add.</p>

**Comment by phadej** [comment]
<p>This isn’t really a solution. It’s just a promise it will be eventually implemented. (So far, it isn’t).</p>

**Comment by dreamer** [comment]
<p>Restarting whole workflows does not work when the event was generated by external contributor sending pull request.</p>

**Comment by chris-rutkowski** [comment]
<p>It has been more than a year now since this “solution”. What is in your backlog is not a solution, it’s a todo.</p>

**Comment by daroga0002** [comment]
<p>seems now it is possible<br>
</p><aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/20227477/181094701-7716dbf1-e968-4894-b5a8-8c687a4c2dd5.png" class="site-icon" width="32" height="32">
      <a href="https://github.blog/changelog/2020-05-26-github-actions-re-run-successful-jobs/" target="_blank" rel="nofollow noopener">The GitHub Blog</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://user-images.githubusercontent.com/20227477/181094702-7bb51917-3d54-45dd-a681-bb2c8c353e23.png" class="thumbnail" width="690" height="362" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/5c656d2360fabf07d147ab889b97ad728a68702f_2_690x362.png, https://aws1.discourse-cdn.com/github/original/2X/5/5c656d2360fabf07d147ab889b97ad728a68702f.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/5/5c656d2360fabf07d147ab889b97ad728a68702f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/5c656d2360fabf07d147ab889b97ad728a68702f_2_10x10.png"></div>

<h3><a href="https://github.blog/changelog/2020-05-26-github-actions-re-run-successful-jobs/" target="_blank" rel="nofollow noopener">GitHub Actions: Re-Run Successful Jobs The GitHub Blog</a></h3>

<p>GitHub Actions: Re-Run Successful Jobs</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by stephanosio** [comment]
<aside class="quote no-group" data-username="daroga0002" data-post="6" data-topic="16145">
<div class="title">
<div class="quote-controls"></div>
 daroga0002:</div>
<blockquote>
<p>seems now it is possible</p>
</blockquote>
</aside>
<p>No, it still does not allow re-triggering/running individual jobs.</p>

**Comment by stephanosio** [comment]
<p>This feature is absolutely critical when you have many long-running jobs in a workflow.</p>
<p>For example, if you take a look at the following link:<br>
</p><aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/stephanosio/zephyr-crosstool-ng/actions/runs/119630859" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/11345418/181094705-8c99d0d8-79af-4687-a728-3303425244c8.jpeg" class="thumbnail onebox-avatar" width="400" height="400">

<h3><a href="https://github.com/stephanosio/zephyr-crosstool-ng/actions/runs/119630859" target="_blank" rel="nofollow noopener">stephanosio/zephyr-crosstool-ng</a></h3>

<p>Zephyr fork of crosstool-NG - for building Zephyr GNU toolchains - stephanosio/zephyr-crosstool-ng</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I have to re-trigger all 42 jobs just because a few jobs randomly failed (source code fetch error) and some macOS jobs got randomly cancelled (yes, this inexplicable event happens a lot for some reason).</p>

**Comment by michaelosthege** [comment]
<p>In <a href="https://github.com/pymc-devs/pymc3" rel="nofollow noopener">PyMC3</a> we use a GitHub action for releasing to PyPI. At the end of the pipeline we test the installation, but sometimes this step fails.</p>
<p>Re-running the entire pipeline is not an option, because re-releasing the same version number is not allowed…</p>

**Comment by rgerhards** [comment]
<p>I would like to add that this is absolutely critical. We sometimes have flakes in our workflow. I was in the middle of migrating over from our buildbot CI, when I came across this problem. Looks like we need to stay with buildbot until this is implemented. We have over 30 jobs, some of them running close to an hour.</p>

**Comment by rgerhards** [comment]
<p>I think I found a workaround here. I have split my matrix build to multiple yaml files, each one with a different name but the same trigger. Each of them contain only a single run. As it looks, this enables me to re-run jobs individually. I do so by selecting “re-run all jobs”, where “all” is now always exactly one.</p>
<p>The price you need to pay is that you have some code duplication and cannot use the matrix feature. For me, this is acceptable as the CI is actually done by a script and shared code outside of that script is relatively minimal.</p>
<p>It would of course be much better if there would be the ability to re-run individual jobs inside the same yaml file.</p>

**Comment by greenfrontend** [comment]
<p><a class="mention" href="/rgerhards">@rgerhards</a> Your workaround is working! Thanks a lot!</p>
